pandas/core/strings/accessor.py

```diff
@@ -316,7 +317,7 @@ def cons_row(x):
             # This is a mess.
             dtype: Optional[str]
             if self._is_string and returns_string:
-                dtype = "string"
+                dtype = self._orig.dtype
```

Comment:
special case here for partition/rpartition/split where the array method returns an object array